.github/workflows/build-test.yml

```diff
@@ -32,23 +32,21 @@ jobs:
         working-directory: v2/
 
       - name: Build
-        run: go build .
-        working-directory: v2/cmd/nuclei/
+        run: bash build.sh retry 3 make build
+        working-directory: v2/
 
       - name: Test
         env:
           GITHUB_TOKEN: "${{ secrets.GITHUB_TOKEN }}"
-        run: go test ./...
+        run: bash build.sh retry 3 make test
         working-directory: v2/
 
       - name: Integration Tests
         env:
           GH_ACTION: true
           GITHUB_TOKEN: "${{ secrets.GITHUB_TOKEN }}"
-        run: |
-          chmod +x run.sh
-          bash run.sh ${{ matrix.os }}
-        working-directory: integration_tests/
+        run: bash build.sh retry 3 make integration 
+        working-directory: v2/
```

Comment:
integration test sometimes fails due to interactsh timeouts etc ref: https://github.com/projectdiscovery/nuclei/actions/runs/3620448423/jobs/6102848991
but not sure if retrying all integration tests is good idea. I think we can implement something granular 
- we already retry a failed test in verbose mode Ref: https://github.com/projectdiscovery/nuclei/blob/1467815ca333d47fc3a6b50c62b6f57524bd7e47/v2/cmd/integration-test/integration-test.go#L64-L75

I think we can simply add a recursion or loop here with retry logic here

Reply:
but got your point as well, we can handle in code as well.